Require Base URL & handle parser errors

aimee-gm opened the pull request:
Fixed #9.

Also does the following:

* Upgarde to node 18, which is current LTS.
* Catch, and report to the user, any errors thrown by `microformats-parser`
* Upgrade `microformats-parser` to `v1.5.2`

aimee-gm commented on views/index.html.ejs:
```diff
@@ -92,20 +90,26 @@
       </form>
 
-      <hr>
+      <hr />
```
These were prettier

aimee-gm commented on index.js:
```diff
@@ -6,21 +6,29 @@ const app = express();
 function getDependencyVersion(dependencyName) {
-  const fs = require('fs');
-  const lockfile = require('@yarnpkg/lockfile');
+  const fs = require("fs");
```
These were prettier

Zegnat commented:
I see that this bumped some dependencies (nice!) and with that even brought in some completely new dependencies (ugh). These seem to be busboy and streamsearch, both being downstream from undici. Node 18 bundles undici (check `process.versions.undici`) for [its native `fetch`](https://nodejs.org/dist/latest-v18.x/docs/api/globals.html#fetch). Could we drop undici completely and losen some more on external dependencies?

The only thing undici seems to be doing for us right now is allow the configuration of `maxRedirections`. But I am not sure how important it actually is to overwrite that from the built-in default.

On another note: could we add Prettier to the `devDependencies` if we are doing automatic formatting? As a fellow Prettier user, I would like that 😉 I am a lot less against development dependencies than shipped dependencies.

aimee-gm commented:
@Zegnat thanks for this, lots to do!

For now, I've done the following:

* Added a `.nvmrc` for anyone using a node version manager
* Added prettier. At the moment, the only thing to enforce this has been the addition of `lint-staged` and `husky` to run prettier in pre-commit hooks.
* Replaced `undici` with the native `fetch` implementation. Let's use the default redirect behaviour and do something about it if it's a problem.

aimee-gm commented:
> I am a lot less against development dependencies than shipped dependencies.

Same here, hence why `microformats-parser` only has the one dependency

Zegnat commented:
Nice! 👏 

I had not heard about lint-staged. Probably going to have a look at that later. Though I am not a big fan of things that depend on git hooks (like husky). In general I believe the types of things people most often run in hooks to actually belong within CI.

aimee-gm commented:
Normally I would have both CI & Git hooks (always annoying to commit & push and for CI to fail for something that could have been auto-fixed in a pre-commit hook)...

But, given how bare-bones this repo is at the moment, I've modified it to just be a GitHub action check, limited to only prettier. I do plan on adding some simple tests, so this can be extended.

I would be keen - for now - not to add anymore to this PR, and perhaps to improve this app in the near future.

aciccarello commented:
Some of the version updates here would be superseded by #10

Zegnat commented:
I have brought in @aciccarello’s version bumps from #12 into this branch, keeping credit intact through `Co-authored-by`. I then did one more push to make sure our other dependencies were up-to-date as well.

With that I actually think we were all already happy to get this merged, so I am just going to do that!

Thanks everyone.
